### ISSUE TITLE
Can PackageData also support a plain js object 

### ISSUE THREAD

**Issue by dkebler** [raineorshine/npm-check-updates#907]
using ncu programatically and I already have a package file as a js object (came from a yml file) and I want to pass that to `ncu.run` but `PackageData` option expects raw json stream from a file.  Can it be enhanced to check if what is passed is a plain object and if so just use it instead of trying to json parse it? thx. 

like so
```javascript
...
 let [err2, tpkg] = await to(ryml(desc('package.yml')))
  if (err2) tpkg = {}

  const upgraded = await ncu.run({
    // Pass any cli option
    packageData: tpkg,
    upgrade: true,
    // Defaults:
    // jsonUpgraded: true,
    // silent: true,
  })

```

**Comment by raineorshine** [comment]
Sure!

Added and published in `v11.7.0`

**Comment by dkebler** [comment]
great!   I have 11.7.1 installed.

only i'm having an issue.  The object I am passing is throwing an error.   Globby is complaining.  Not sure why globby is even being called if I'm passing the object instead of a filename.
 
https://github.com/raineorshine/npm-check-updates/blob/857b825f0f57f3d23ca1c53e2a8f12b5ec8b4976/lib/index.js#L494
https://github.com/sindresorhus/globby/blob/94e192c7b73fa76f9a95131fe4714aec94731ebe/index.js#L41


so passing I'm this object
```
{
  main: 'src/index.js',
  type: 'module',
  scripts: {
    test: './node_modules/.bin/mocha --timeout 30000',
    'test:dev': "UCI_ENV=dev ./node_modules/.bin/nodemon --exec './node_modules/.bin/mocha --timeout 30000' || exit 0",
    'test:dev:trace': "UCI_LOG_LEVEL='trace' npm run test:dev",
    'test:dev:warn': "UCI_LOG_LEVEL='warn' npm run test:dev",
    'test:log': 'UCI_ENV=pro UCI_LOG_PATH=./test/test.log 0 npm run test || exit 0'
  },
  author: 'David Kebler',
  license: 'MIT',
  engines: { node: '>=16.0.0' },
  devDependencies: {
    chai: '^4.3.4',
    mocha: '^9.0.1',
    nodemon: '^2.0.9',
    rxjs: '^7.1.0'
  },
  keywords: [
    'node.js',
    'type',
    'typecast',
    'cast',
    'typeof',
    'typechecker',
    'variables'
  ],
  name: '@uci-utils/type',
  version: '0.6.4',
  description: 'A variable type check and casting utility - extension of typechecker package plus type casting',
  repository: {
    type: 'git',
    url: 'git+https://github.com/uCOMmandIt/uci-utils.git'
  },
  bugs: { url: 'https://github.com/uCOMmandIt/uci-utils/issues' },
  homepage: 'https://github.com/uCOMmandIt/uci-utils#readme',
  dependencies: {
    '@uci-utils/to-boolean': '^0.2.0',
    'is-observable': '^2.1.0',
    'observable-to-promise': '^1.0.0',
    typechecker: '^7.17.0'
  }
}
```

getting this error
```
fatal TypeError: Patterns must be a string or an array of strings
    at assertPatternsInput (/mnt/AllData/hacking/active/package/node_modules/npm-check-updates/node_modules/globby/index.js:16:9)
    at generateGlobTasks (/mnt/AllData/hacking/active/package/node_modules/npm-check-updates/node_modules/globby/index.js:41:2)
    at AsyncFunction.module.exports.sync (/mnt/AllData/hacking/active/package/node_modules/npm-check-updates/node_modules/globby/index.js:139:20)
    at getAnalysis (/mnt/AllData/hacking/active/package/node_modules/npm-check-updates/lib/index.js:493:25)
    at Object.run (/mnt/AllData/hacking/active/package/node_modules/npm-check-updates/lib/index.js:565:42)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async file:///mnt/AllData/hacking/active/package/index.js:73:20
```

using this function
```
 const upgradeable = (pkg) => ncu.run({
    // Pass any cli option
    packageFile: pkg || {},
    upgrade: true
    // Defaults:
    // jsonUpgraded: true,
    // silent: true,
  })
```
calling like this.
```
  let [err, bpkg] = await to(ryml(config('package.yml')))
  if (err) bpkg = {}
  // console.log(filter(bpkg, ['engines', 'keywords']))
  let [err2, tpkg] = await to(ryml(dest('package.yml')))
  if (err2) tpkg = {}
  tpkg = Object.assign(bpkg, tpkg)
  console.log('template\n')
  console.dir(tpkg)    
  await upgraded(tpkg)
```

**Comment by dkebler** [comment]
THX!   Just got around to trying it out and it works great!  

(a couple hair pulling minutes until I realized I needed to change key to `packageData` from `packageFile`)

So I'm closing this.
